### PR TITLE
DOCS Clarify security process, introduce CVE and CVSS

### PR DESCRIPTION
**Pull request by chillu**
Moved the guts to "making a core release", since it's only really relevant to that audience.
There's more work to do around making security and non-security releases the same (less special handling),
but I think this is a good start.

[ci-skip]

Thanks for contributing, you're awesome! :star:
Please describe expected and observed behaviour, and what you're fixing.
For visual fixes, please include tested browsers and screenshots.
Search for related existing issues and link to them if possible.
Please read https://docs.silverstripe.org/en/contributing/code/

### CODE REVIEW

File: docs/en/05_Contributing/04_Release_Process.md [hunk 1]
```diff
@@ -128,16 +128,11 @@ In the event of a confirmed vulnerability in our
...
-   Example: `SS-2013-001` would be the first of the year `2013`.
-   Additionally, [CVE](http://cve.mitre.org) numbers are accepted.
-*  Halt all other development as long as is needed to develop a fix, including patches against the current and one
-previous major release (if applicable).
-* Pre-announce the upcoming security release to aprivate pre-announcement mailing list of important stakeholders (see below).
-*  We will inform you about resolution and [announce](https://forum.silverstripe.org/c/releases) a 
+* Acknowledge to the reporter that we’ve received the report and that a fix is forthcoming.
+  We’ll give a rough timeline and ask the reporter to keep the issue confidential until we announce it.
+* Assign a [CVE identifier](https://cve.mitre.org) to the issue.
+* For "high" and "critical" issues: Pre-announce the upcoming security release to aprivate pre-announcement mailing list of important stakeholders (see below).
```
Comment (sminnee) [inline docs/en/05_Contributing/04_Release_Process.md]: Perhaps say `For "high" and "critical" issues (CVSS > 7):` here. You can figure that out from the table below but it'd make the text more easily understood if you were familiar with CVSS scores.

**Comment by sminnee** [comment]
Make a couple of comments, but looks good.

File: docs/en/05_Contributing/04_Release_Process.md (hunk 1 above)
Comment (chillu) [reply docs/en/05_Contributing/04_Release_Process.md]: Sweet, fixed.

**Comment by chillu** [comment]
I only see one comment, not a couple - but addressed this one thing

**Comment by chillu** [comment]
OK, I've got two 👍 from core committers, merging.

**Pull request merged**